### PR TITLE
Add optional terminal numbers to toggle term

### PR DESCRIPTION
**Pull request by kraxli**
enhance #300 by providing (optional) terminal number

e.g. key stroke 4tp with open python in terminal  number 4

### CODE REVIEW

**Comment by mehalter** [comment]
Thanks for the contribution! @kraxli  could you please rebase your commits into a single commit. We are not allowing any merge commits in main to keep a linear commit log.

File: lua/core/utils.lua [hunk 1]
```diff
@@ -164,11 +164,15 @@ function M.toggle_term_cmd(term_details)
   if type(term_details) == "string" then
     term_details = { cmd = term_details, hidden = true }
   end
-  local cmd = term_details.cmd
-  if M.user_terminals[cmd] == nil then
-    M.user_terminals[cmd] = require("toggleterm.terminal").Terminal:new(term_details)
+  local term_key = term_details.cmd
+  if vim.v.count > 0 and term_details.count == nil then
+    term_details.count = vim.v.count
+    term_key = term_details.cmd .. vim.v.count
```
Comment (mehalter) [inline lua/core/utils.lua:170]: Let's replace this line with `term_key = term_key .. vim.v.count`

**Review by mehalter — CHANGES_REQUESTED** [review]
Looks pretty great! I suggested one line change and other than that I just need you to rebase the commits back down to a single commit instead of having merge and revert commits.

**Review by mehalter — CHANGES_REQUESTED** [review]
Thanks for rebasing, its quite close. Can you please drop the unrelated commit that changes bufferline autocommands. Seems to be a problem with the Rebase.

**Comment by kraxli** [comment]
thanks, I followed your  suggestion. Hope it fits now :-)

**Review by mehalter — APPROVED** [review]
LGTM

**Pull request merged**